libpod/oci_conmon_linux.go

```diff
@@ -575,13 +575,36 @@ func (r *ConmonOCIRuntime) HTTPAttach(ctr *Container, httpConn net.Conn, httpBuf
 	}
 }
 
+// isRetryable returns whether the error was caused by a blocked syscall or the
+// specified operation on a non blocking file descriptor wasn't ready for completion.
+func isRetryable(err error) bool {
+	if errno, isErrno := errors.Cause(err).(syscall.Errno); isErrno {
+		return errno == syscall.EINTR || errno == syscall.EAGAIN
+	}
+	return false
+}
+
+// openControlFile opens the terminal control file.
+func openControlFile(ctr *Container) (*os.File, error) {
+	controlPath := filepath.Join(ctr.bundlePath(), "ctl")
```

Comment:
Is this correct? The bundle path differs for exec and non-exec

Reply:
This actually only works for attach, not exec attach - we need to use `filepath.Join(ctr.execBundlePath(sessionID), "ctl")` for exec

Reply:
thanks for noticing it!  Surprisingly Podman works also with the  `filepath.Join(ctr.bundlePath(), "ctl")` version